ACCESS-OM3 Feedback on How to Run (#1043)
ACCESS-Hive/access-hive.github.io: docs/models/run_a_model/run_access-om3.md

----------------------------------------------------------------------------------------

## Realise an experiment

Once the configuration, and any modifications, are functioning as expected, the configuration can be used to run an experiment. 
To realise an experiment, conduct a series of short runs of the configuration until the 
desired length of the experiment is reached.

If the previous run has [finished succesfully](#model-log-files), then extend it for for another run of the configured run length by executing:

    payu sweep && payu run

To automatically start multiple runs consecutively, use the `-n` flag to request the number of runs.

    payu run -n N

Where N is the number of runs desired of the configured run length. _Payu_ will submit a run to the queue on gadi, and when it has finished succesffuly
payu submits the next run until a total of N consecutive runs is reached.

----------------------------------------------------------------------------------------

> 

Reviewer: Thank you for this explanation. 
I like the idea of this paragraph, and I think it would be good added similarly to how it's done in the [How to run ACCESS-OM2 page](https://docs.access-hive.org.au/models/run-a-model/run-access-om2/#run-configuration-for-more-than-5-years)

Not sure what is the recommended limit on the run length for ACCESS-OM3, but if it's 1 year, we can use that.

This means this paragraph would be moved under the `Change run length and restart period`.

Also, I suggest a few modifications:
````suggestion
#### Run configuration for more than 1 year

Once a run has [successfully completed](#model-log-files), it can be extended for another [run length](https://github.com/ACCESS-NRI/ACCESS-Hive-Docs/pull/1043/files#change-run-length-and-restart-period) to continue from where it left off by executing:

```
payu sweep && payu run
```

Alternatively, to automatically execute multiple consecutive runs, each having a [run length](#change-run-length-and-restart-period) duration, use the `-n` flag:

```
payu run -n <number-of-runs>
```
by replacing <number-of-runs> with the desired number of consecutive runs.

_Payu_ will submit the first run to the Gadi queue and, upon successful completion, it will automatically submit the next run, continuing until the total of `<number-of-runs>` runs is reached.
----------------------------------------------------------------------------------------
````

Author: There's no recommended run length really. It depends on the configuration

Author: > Technically, instead of executing `payu run -n N`, I could also change the run length so that my experiment would run the same amount of time only by executing `payu run`.

Sadly this crashes with insufficient walltime

Reviewer: Good point. 
Then, I think similarly to what we do in [Run ACCESS-OM2](https://docs.access-hive.org.au/models/run_a_model/run_access-om2/#change-run-length) with a warning, we should advise people not to increment the run length to more than a set value (maybe we can set 1 year?) to avoid problems. 
If they want to run an experiment for more years, they should follow instructions on how to do so (in the following sub-paragraph - i.e., `payu run -n ...`)

## Edit {{ model }} configuration

This section describes how to modify an {{ model }} configuration.<br>

> The modifications discussed in this section can change how the model software and it's [components](model components) are configured, or the way {{ model }} is run by _payu_, 

Author: Oops - thanks. Ill just keep it simple and use [model components]

More details on model and _payu_ configuration are found in the 
[Configurations Overview](https://access-om3-configs.access-hive.org.au/configurations/Overview/) section of {{ model }} config docs.

> ### Edit model component configuration

Reviewer: > I agree this is confusing, but I can't see a good way to say "to extend a model run it multiple times, but to shorten a run change the model config" without getting into some details

I think we're going to the right direction with that and the instructions for ACCESS-OM2 already explain it in a reasonably good way.

> Similarly, explaining payu compared to the model helps with explaining why there is both options for how frequently restarts are saved compared with restart_freq set for payu to prune by

I understand this might get confusing for people if you don't explain the difference between a model configuration and _payu_'s configuration. However, I think the objective in general is for users to:
- know how to carry out specific tasks (e.g., changing run length)
- be advised of some features that may be helpful for them (e.g., restart pruning)

This could be achieved also without getting into the details of which configuration to modify

----------------------------------------------------------------------------------------

> ## Run an experiment

Author: Level 2 is intentional here

> 

Author: I've linked to the faq instead

To conduct an experiment, do a series of runs until the desired length of the 
experiment is reached.

To run a configuration for more than the configured run length, use the `-n` option:

    payu run -n <number-of-runs>

This will run {{ model }} `number-of-runs` consecutive times for the configured 
run length. This way, the *total experiment length* will be `run-length * number-of-runs`. 

For example, to run a configuration for a total of 10 years with the default run 
length (1 year), the `number-of-runs` should be set to `10`:

    payu run -n 10

----------------------------------------------------------------------------------------

## Edit {{ model }} configuration

This section describes how to modify an {{ model }} configuration.<br>
The modifications discussed in this section can change how the model software and 
it's [model components] are configured, or the way {{ model }} is run by _payu_.

The `config.yaml` file located in the _control_ directory is the _payu_ configuration
file, which controls the configuration of the experiment manager. It contains several 
parts, some of which it is more likely will need modification, and others which are 

> rarely changed.

Reviewer: ```suggestion
file, which controls the configuration of the experiment manager. It consists of several parts, some of which are frequently modified, whereas others are rarely changed.
```

> More details on model and _payu_ configuration specific to {{ model }} are found in the 

Reviewer: ```suggestion
More details on model and _payu_ configurations specific to {{ model }} are found in the 
```

Preset `diag_table` files, along with the YAML configuration files used to generate them, can be found in the `diagnostic_profiles` subdirectory of the _control_ directory. To create and use a custom `diag_table`, follow the instructions in `diagnostic_profiles/README.md`.

!!! warning
    MOM6 provides the ability to vertically remap diagnostics onto user-defined vertical coordinates, including density coordinates (check [MOM6 vertically remapped diagnostics documentation](https://mom6.readthedocs.io/en/main/api/generated/pages/Diagnostics.html#vertically-remapped-diagnostics) for more information). Remapping to density coordinates can add substantially to the runtime of the model. The default `diag_table` used by {{ model }} includes diagnostics remapped to density coordinates. These should be removed for performance reasons if they are not needed.

### Change run length and restart period

> The run length, or model duration, can be reduced to minimise resource consumption when debugging and return faster feedback on changes.

Reviewer: ```suggestion
The run length (also called model duration) can be reduced to minimise resource consumption when debugging and return faster feedback on changes.
```

         stop_option = nmonths
         ...

> To run a model for longer than the default run length, conduct multiple runs rather than increasing run length. See [Run an experiment](#run-an-experiment).

Reviewer: Here I would use a warning similarly to How to run ACCESS-OM2:
```suggestion
While the model run length can be shortened, it is recommended not to increase that to more than 1 year. To run a model for longer than 1 year, see [Run configuration for more than 1 years](#run-configuration-for-more-than-1-years).
```
The `#run-configuration-for-more-than-1-years` paragraph would need to be added, as mentioned in this comment.

#### Create a custom {{ model }} build
All the executables needed to run {{ model }} are pre-built using _Spack_.<br>
To customise {{ model }}'s build, including changes in the source code of one of its components, refer to [Modify and build an ACCESS model's source code](/models/run-a-model/build_a_model#{{model|lower}}).

### Edit _payu_ configuration

_Payu_ is the experiment manager used for {{ model }}. It tracks input and configuration files used and organises model output. 
It also provides a common interface to several ACCESS models.
The `config.yaml` file located in the _control_ directory is the _payu_ configuration, which controls the configuration of the experiment manager. It contains several parts, some of which it is more likely will need modification, and others which are rarely changed without having a deep understanding of how the model is configured.

To find out more about configuration settings for the `config.yaml` file, refer to [how to configure your experiment with payu](https://payu.readthedocs.io/en/latest/config.html).

> 

Author: In general I reordered to follow the OM2 page, which has this effect

restart_freq: '50YS'
```

For more information, check [_payu_ Configuration Settings documentation](https://payu.readthedocs.io/en/latest/config.html#model).

#### Input, forcing and exectuable configuration {: .no-toc }

This section tells _payu_ which driver to use for the main model configuration (`access-om3`) and the location of the model 

> executable and all input files. These input files capture data needed for the experiment to be run, including grids, bathymetry, 

Author: Was this a typo?, i don't understand this suggestion

restart_freq: '50YS'
```

For more information, check [_payu_ Configuration Settings documentation](https://payu.readthedocs.io/en/latest/config.html#model).

#### Input, forcing and exectuable configuration

This section tells _payu_ which driver to use for the main model configuration (`access-om3`) and the location of the model 

> executable and all input files. These input files capture data needed for the experiment to be run, including grids, bathymetry, 

Reviewer: ```suggestion
executable and input files. These input files capture data needed for the experiment to be run, including grids, bathymetry, 
```
